src/OnboardingSPA/components/Drawer/DrawerPanel/DesignColors.js

```diff
@@ -1,15 +1,378 @@
+	const [isAccordionClosed, setIsAccordionClosed] = useState(true);
+
+	const [primaryColor, setPrimaryColor] = useState();
+	const [tertiaryColor, setTertiaryColor] = useState();
+	const [secondaryColor, setSecondaryColor] = useState();
+	const [backgroundColor, setBackgroundColor] = useState();
+	const [colorPickerCalledBy, setColorPickerCalledBy] = useState('');
+
+	const { storedPreviewSettings, currentData } = useSelect(
+		(select) => {
+			return {
+				storedPreviewSettings:
+					select(nfdOnboardingStore).getPreviewSettings(),
+				currentData:
+					select(nfdOnboardingStore).getCurrentOnboardingData(),
+			};
+		},
+		[]
+	);
+
+	const { updatePreviewSettings, setCurrentOnboardingData } =
+		useDispatch(nfdOnboardingStore);
+
+	const colorPalettes = {
+		'calm': [
+			'#C7DBFF',
+			'#E6EBEE',
+			'#1A4733',
+		],
+		'cool': [
+			'#C7DBFF',
+			'#EDF7FE',
+			'#21447B',
+		],
+		'warm': [
+			'#FFEDED',
+			'#FEF7E8',
+			'#7A3921',
+		],
+		'radiant': [
+			'#C7F0FF',
+			'#FEF4FB',
+			'#63156A',
+		],
+		'bold': [
+			'#F2A3D6',
+			'#FFFBF5',
+			'#09857C',
+		],
+		'retro': [
+			'#F2E6A2',
+			'#F5FFFF',
+			'#096385',
+		],
+		'professional': [
+			'#A2C1F2',
+			'#F5FAFF',
+			'#669933',
+		],
+	}
+
+	async function setThemeColorPalette(colorStyle, selectedColorsTemp = selectedColors, globalStylesTemp = globalStyles) {
+		const isCustomStyle = colorStyle === 'custom';
+		let primaryColorTemp = selectedColorsTemp?.color[0].color ?? null;
+		let secondaryColorTemp = selectedColorsTemp?.color[1].color ?? null;
+		let tertiaryColorTemp = selectedColorsTemp?.color[2].color ?? null;
+		let backgroundColorTemp = selectedColorsTemp?.color[3].color ?? null;
+
+		let selectedGlobalStyle = globalStylesTemp;
+		let selectedThemeColorPalette = selectedGlobalStyle?.settings?.color?.palette?.theme;
+
+		if (colorStyle && selectedThemeColorPalette) {
+			for (let idx = 0; idx < selectedThemeColorPalette.length; idx++) {
+				switch (selectedThemeColorPalette[idx]?.slug) {
+					case 'primary':
+						if (isCustomStyle && primaryColorTemp)
+							selectedThemeColorPalette[idx].color = primaryColorTemp;
+						else
+							selectedThemeColorPalette[idx].color = colorPalettes[colorStyle][2];
+						break;
+					case 'secondary':
+						if (isCustomStyle && secondaryColorTemp)
+							selectedThemeColorPalette[idx].color = secondaryColorTemp;
+						else
+							selectedThemeColorPalette[idx].color = colorPalettes[colorStyle][1];
+						break;
+					case 'tertiary':
+						if (isCustomStyle && tertiaryColorTemp)
+							selectedThemeColorPalette[idx].color = tertiaryColorTemp;
+						else
+							selectedThemeColorPalette[idx].color = colorPalettes[colorStyle][0];
+						break;
+					case 'background':
+						if (isCustomStyle && backgroundColorTemp)
+							selectedThemeColorPalette[idx].color = backgroundColorTemp;
+						else
+							selectedThemeColorPalette[idx].color = '#ffffff';
+						break;
+				}
+			}
```

Comment:
We can eliminate such logic of using for loops and iteration variables if we can move to a more objective data structure.

Reply:
This is how the PreviewSettings is structured already and we have to do it this way.